students/danwoj/dlpscan/dlpscan.py

```diff
@@ -0,0 +1,322 @@
+    return bin_octet
+
+def sn_calculation(ipa_value, snm_value):
+# This function takes the binary lists of both the subnet mask
+# and the IP address and returns the subnet in CIDR notation
+    cidr = 0
+    bin_subnet_address = []
+# As I look at the next three variables I feel like I'm sort of 
+# cheating by defining these empty values like this and setting 
+# my range below that to four but I was in a pinch 
+# and it seemed to work so ¯\_(ツ)_/¯.
+    final_subnet_address = ''
+    octets = ['','','','']
+    dec_octets = ['','','','']
+    for i in range(4):
+        for inum, snum in zip(ipa_value[i], snm_value[i]):
+            if snum == '1':
+                cidr += 1
+                bin_subnet_address.append(inum)
+                octets[i] += inum
+            else:
+                bin_subnet_address.append('0')
+                octets[i] += '0'
+            dec_octets[i] = int(octets[i], 2)
+        final_subnet_address += str(dec_octets[i])
+        if i < 3:
+            final_subnet_address += '.'
+    final_subnet_address += '/'
+    final_subnet_address += str(cidr)
+    return final_subnet_address
+
+def subnet_calculator(ip_addr, subnet_mask):
+# This function calculates a network subnet based on a valid IP 
+# address and its subnet mask
+
+    # These two lines take the IP address and subnet mask strings 
+    # and turn them into an int list of octet values
+    ip_addr_list = ip_addr.split('.')
+    subnet_mask_list = subnet_mask.split('.')
+    # This converts the IP address into binary form
+    ipa_compare = ip_oct_bin(ip_addr_list)
+    # This converts the subnet mask into binary form
+    snm_compare = ip_oct_bin(subnet_mask_list)
+    # This takes the two binary values and calculates the subnet
+    subnet = sn_calculation(ipa_compare, snm_compare)
+    return subnet
+
+class Net_data:
+# This class defines the network data (IP, subnet mask, and default gateway) for the host running the application
+    def __init__(self):
+        self.ip
+        self.subnet
+        self.gateway
+
+    def ip(self):
+        ip = (([ip for ip in socket.gethostbyname_ex(socket.gethostname())[2] if not ip.startswith("127.")] or [[(s.connect(("8.8.8.8", 53)), s.getsockname()[0], s.close()) for s in [socket.socket(socket.AF_INET, socket.SOCK_DGRAM)]][0][1]]) + ["no IP found"])[0]
+
+        return ip
+
+    def subnet(self):
+        Popen_command = ''
+        subnet = sub.Popen("ifconfig | grep {} | awk -F 'Mask:' '{{print $2}}'".format(self.ip()), stdout = sub.PIPE, shell = True)
+        (output, err) = subnet.communicate()
+        subnet = (str(output, "utf8").split("\n")).pop(0)
+
+        return subnet
+
+    def gateway(self):
+        default_gate = sub.Popen("ip route show | grep default | awk -F ' ' '{print $3}'", stdout = sub.PIPE, shell = True)
+        (output, err) = default_gate.communicate()
+        default_gate = (str(output, "utf8").split("\n")).pop(0)
+
+        return default_gate
+
+def sudo(s,password):
```

Comment:
I would not call this function sudo. While not a python keyword, your program is doing enough terminal commands, that I would be worried about users (or other developers) becoming confused about its purpose.